mp2i/cogs/sanctions.py

```diff
@@ -127,8 +223,234 @@ async def unwarn(self, ctx, id: int) -> None:
+                return
+        staff = entry.user  # renommage pour meilleure compréhension
+
+        def insert_in_database(type, duration):
+            self.__register_sanction_in_database(
+                staff=entry.user.id,
+                member=entry.target.id,
+                guild=entry.guild.id,
+                date=datetime.now(),
+                type=type,
+                duration=duration,
+                reason=entry.reason
+            )
+            logger.info(f"{staff.name} ({staff.id}) {type} {user.name} ({user.id}).")
+
+        if entry.action == AuditLogAction.ban:
+            insert_in_database("ban", None)
+            await handle_log_ban(user, staff, entry.reason)
+
+        elif entry.action == AuditLogAction.unban:
+            insert_in_database("unban", None)
+            await handle_log_unban(user, staff)
+
+        # Doit être étrangement avant la condition de TO sinon ne s'applique pas
+        elif entry.action == AuditLogAction.member_update and (
+            entry.before.timed_out_until and not entry.after.timed_out_until
+        ):
+            insert_in_database("unto", None)
+            await handle_log_unto(user, staff)
+
+        elif entry.action == AuditLogAction.member_update and (
+            not entry.before.timed_out_until
+            and entry.after.timed_out_until
+            or entry.before.timed_out_until
+            and entry.before.timed_out_until < entry.after.timed_out_until
+        ):
+            end_of_sanction = entry.after.timed_out_until.timestamp()
+            insert_in_database("to", int(ceil(end_of_sanction - datetime.now().timestamp())))
+            # +60 indique la minute qui suit, mieux vaut large que pas assez
+            await handle_log_to(user, staff, entry.reason, int(end_of_sanction + 60))
+
+    class WarnModal(Modal, title="Avertir un member"):
+        def __init__(self, sanction, member):
+            super().__init__()
+            self.sanction = sanction
+            self.member = member
+            self.add_item(
+                TextInput(
+                    label = 'Raison',
+                    placeholder = "Votre comportement ne sied guère à l'esprit du serveur...",
+                    max_length = 3096,
+                    style = TextStyle.paragraph,
+                    required = True
+                )
+            )
+            self.add_item(
+                TextInput(
+                    label = "Envoi d'un message privé",
+                    placeholder = '`non` => ne pas envoyer, autre chose => oui',
+                    max_length = 10,
+                    required = False
+                )
+            )
+
+        async def on_submit(self, interaction: discord.Interaction):
+            await interaction.response.defer()
+            reason = self.children[0].value
+            send_dm = self.children[1].value != "non"
```

Comment:
```suggestion
            send_dm = self.children[1].value.strip().lower() != "non"
```
Ça coûte pas grand chose et c'est toujours clair